base_layer/core/src/base_node/proto/chain_metadata.rs

```diff
@@ -50,10 +53,17 @@ impl TryFrom<proto::ChainMetadata> for ChainMetadata {
         } else {
             height_of_longest_chain.saturating_sub(metadata.pruned_height)
         };
-
+        let hash: FixedHash = match metadata
+            .best_block
+            .ok_or_else(|| "Best block is missing".to_string())?
+            .try_into()
+        {
+            Ok(v) => v,
+            Err(e) => return Err(format!("Malformed best block: {}", e)),
+        };
```

Comment:
nit: manual implementation of map_err
```suggestion
        let hash: FixedHash = match metadata
            .best_block
            .ok_or_else(|| "Best block is missing".to_string())?
            .try_into()
       .map_err(|e| Err(format!("Malformed best block: {}", e))?;
```